internal/pkg/http/client.go

```diff
@@ -73,17 +74,22 @@ func NewClient(host string, insecure bool, timeoutMilliseconds int, protocol Pro
 }
 
 // SendRequest sends a request to the HTTP server and wraps useful information into a Response object.
-func (c Client) SendRequest(method, path string, headers map[string]string, body io.Reader) response.Response {
+func (c Client) SendRequest(method, path string, headers map[string]string, requestBody *string) response.Response {
 	const respType = "http"
-
+	var body io.Reader
+	if requestBody != nil {
+		body = bytes.NewBufferString(*requestBody)
+	}
+	
 	url := fmt.Sprintf("%s/%s", c.host, strings.TrimLeft(path, "/"))
 	req, err := http.NewRequest(method, url, body)
-
 	if err != nil {
 		log.Printf("Failed to create request: %s %s: %v", method, url, err)
 		return response.Response{Duration: time.Duration(0), Err: err, Type: respType}
 	}
-
+	if req.Body != nil {
```

Comment:
Body must be closed when there is no error and when a body is present. In case of GET there is no Body.